rama-http-backend/src/client/mod.rs

```diff
@@ -440,103 +158,404 @@ where
+
+        #[cfg(not(any(feature = "rustls", feature = "boring")))]
+        /// Add a http request inspector that will run just after the inner http connector
+        /// has finished but before the http handshake
+        pub fn with_jit_req_inspector<I>(
+            self,
+            http_req_inspector: I,
+        ) -> EasyHttpWebClientBuilder<HttpConnector<T, I, I2>, HttpStage> {
+            EasyHttpWebClientBuilder {
+                connector: self.connector.with_jit_req_inspector(http_req_inspector),
+                _phantom: PhantomData,
+            }
+        }
+
+        /// Add a http request inspector that will run just before doing the actual http request
+        pub fn with_svc_req_inspector<I>(
+            self,
+            http_req_inspector: I,
+        ) -> EasyHttpWebClientBuilder<HttpConnector<T, I1, I>, HttpStage> {
+            EasyHttpWebClientBuilder {
+                connector: self.connector.with_svc_req_inspector(http_req_inspector),
+                _phantom: PhantomData,
+            }
+        }
+    }
+
+    type DefaultConnectionPoolBuilder<T, C> = EasyHttpWebClientBuilder<
+        PooledConnector<T, FiFoReuseLruDropPool<C, BasicHttpConId>, BasicHttpConnIdentifier>,
+        PoolStage,
+    >;
+
+    impl<T> EasyHttpWebClientBuilder<T, HttpStage> {
+        /// Use the default connection pool for this [`super::EasyHttpWebClient`] with the default config
+        ///
+        /// This will create a [`FiFoReuseLruDropPool`] using the default limits
+        /// and will use [`BasicHttpConnIdentifier`] to group connection on protocol
+        /// and authority, which should cover most common use cases
+        ///
+        /// If you need different limits you can use [`EasyHttpWebClientBuilder::with_connection_pool()`]
+        ///
+        /// If you need a different pool or custom way to group connection you can
+        /// use [`EasyHttpWebClientBuilder::with_custom_connection_pool()`] to provide
+        /// you own.
+        pub fn with_default_connection_pool<C>(
+            self,
+        ) -> Result<DefaultConnectionPoolBuilder<T, C>, OpaqueError> {
+            self.with_connection_pool(HttpPooledConnectorConfig::default())
+        }
+
+        /// Use the default connection pool for this [`super::EasyHttpWebClient`]
+        ///
+        /// This will create a [`FiFoReuseLruDropPool`] using the provided limits
+        /// and will use [`BasicHttpConnIdentifier`] to group connection on protocol
+        /// and authority, which should cover most common use cases
+        ///
+        /// Use `wait_for_pool_timeout` to limit how long we wait for the pool to give us a connection
+        ///
+        /// If you need a different pool or custom way to group connection you can
+        /// use [`EasyHttpWebClientBuilder::with_custom_connection_pool()`] to provide
+        /// you own.
+        pub fn with_connection_pool<C>(
+            self,
+            config: HttpPooledConnectorConfig,
```

Comment:
Usually I allow this to be `Option` and than inside you would do `unwrap_or_default`, wdyt?